gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJobLauncher.java

```diff
@@ -365,20 +369,51 @@ public Void call() throws Exception {
     return workUnitFile.toString();
   }
 
-  private void waitForJobCompletion() throws InterruptedException {
-    while (true) {
+  private void waitForJobCompletion()  throws InterruptedException {
+    LOGGER.info("Waiting for job to complete...");
+    long timeoutInSeconds;
+    timeoutInSeconds = Long.parseLong(this.jobProps.getProperty(ConfigurationKeys.HELIX_JOB_TIMEOUT_SECONDS,
+        ConfigurationKeys.DEFAULT_HELIX_JOB_TIMEOUT_SECONDS));
+    long endTime = System.currentTimeMillis() + timeoutInSeconds*1000;
+    while (System.currentTimeMillis() <= endTime) {
       WorkflowContext workflowContext = TaskDriver.getWorkflowContext(this.helixManager, this.helixQueueName);
       if (workflowContext != null) {
         org.apache.helix.task.TaskState helixJobState = workflowContext.getJobState(this.jobResourceName);
-        if (helixJobState == org.apache.helix.task.TaskState.COMPLETED ||
-            helixJobState == org.apache.helix.task.TaskState.FAILED ||
-            helixJobState == org.apache.helix.task.TaskState.STOPPED) {
+        if (helixJobState == org.apache.helix.task.TaskState.COMPLETED
+            || helixJobState == org.apache.helix.task.TaskState.FAILED
+            || helixJobState == org.apache.helix.task.TaskState.STOPPED) {
           return;
         }
       }
-
       Thread.sleep(1000);
     }
+    helixTaskDriverWaitToStop(this.helixQueueName, 10L);
+    try {
+      cancelJob(this.jobListener);
+    } catch (JobException e) {
+      throw new RuntimeException("Unable to cancel job " + jobContext.getJobName() + ": ", e);
+    }
+    this.helixTaskDriver.resume(this.helixQueueName);
+    LOGGER.info("stopped the queue, deleted the job");
+  }
+
+  /**
+   * Because waitToStop is not available 0.6.7
+   */
+  private void helixTaskDriverWaitToStop(String workflow, long timeoutInSeconds) throws InterruptedException {
+    this.helixTaskDriver.stop(workflow);
```

Comment:
@htran1 Helix operations can fail if there is ZK connectivity issues.  Do we retry these at a lower level, or should then be retried by the caller?